genesis/process/disabled/messageSignVerifier.go

```diff
@@ -1,15 +1,39 @@
 package disabled
 
+import (
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/genesis"
+)
+
 // MessageSignVerifier represents the message verifier that accepts any message, sign, pk tuple
-type MessageSignVerifier struct {
+// and only verifies public key validity
+type messageSignVerifier struct {
+	kg crypto.KeyGenerator
+}
+
+// NewMessageSignVerifier creates a new message sign verifier - which verifies only the public key validity
+func NewMessageSignVerifier(kg crypto.KeyGenerator) (*messageSignVerifier, error) {
+	if check.IfNil(kg) {
+		return nil, genesis.ErrNilKeyGenerator
+	}
+	return &messageSignVerifier{kg: kg}, nil
 }
 
-// Verify returns always nil
-func (msv *MessageSignVerifier) Verify(_ []byte, _ []byte, _ []byte) error {
+// Verify returns nil if public key is valid
+func (msv *messageSignVerifier) Verify(_ []byte, _ []byte, pubKey []byte) error {
+	if len(pubKey) == 0 {
+		return genesis.ErrEmptyPubKey
+	}
+	_, err := msv.kg.PublicKeyFromByteArray(pubKey)
```

Comment:
Surprisingly, this does not work as expected. Please add messageSignVerifier_test.go containing:
```GO
package disabled

import (
    "testing"

    "github.com/ElrondNetwork/elrond-go/crypto/signing"
    "github.com/ElrondNetwork/elrond-go/crypto/signing/mcl"
    "github.com/stretchr/testify/assert"
)

func TestMessageSignVerifier_Verify(t *testing.T) {
    t.Parallel()

    suite := mcl.NewSuiteBLS12()
    keygen := signing.NewKeyGenerator(suite)

    sv, _ := NewMessageSignVerifier(keygen)

    err := sv.Verify(nil, nil, make([]byte, 97))
    assert.NotNil(t, err)
}
```

Reply:
done